### ISSUE TITLE
Rework perf hw_breakpoint support

### ISSUE THREAD

**Issue by mpe** [linuxppc/issues#38]
The powerpc hw_breakpoint support is the only arch that uses the `arch_unregister_hw_breakpoint()` hook. In there we do:

```
if (bp->ctx && bp->ctx->task && bp->ctx->task != ((void *)-1L))
    bp->ctx->task->thread.last_hit_ubp = NULL;
```

Peterz strongly suggests we should not be using `bp->ctx` in there, and that we should (probably) instead be storing the breakpoint info per-cpu like other arches:

> Indeed, but if there's a preemption point in between setting and using
> that state, the ctx->task pointer might not actually still point to the
> same task. With inherited events the event might get swapped to the next
> task if it has the exact same (inherited) event configuration instead of
> reprogramming the hardware.

The code needs to be reworked & tested.

**Comment by mpe** [comment]
Possible solution from @BenjaminGrayNp1 : https://patchwork.ozlabs.org/project/linuxppc-dev/list/?series=366657

**Comment by mpe** [comment]
Fixed by @BenjaminGrayNp1 in torvalds/linux@53834a0c09252dea7918a9e1788bad880690900b
